### PR TITLE
Fix codecs

### PR DESCRIPTION
**Pull request by askolesov**
Fixed cheqd types registration.
Fixed proto packing/unpacking.

These issues caused panic during nymList export to genesis.

### CODE REVIEW

File: x/cheqd/types/keys.go [hunk 1]
```diff
@@ -27,5 +27,5 @@ const (
 	DidCountKey = "did-count:"
 	// DidNamespaceKey FIXME: Should be `did-namespace:`.
 	// Networks was started with `testnet` value so we need a migration now.
-	DidNamespaceKey = "testnet"
+	DidNamespaceKey = "testnettestnet"
```
Comment (ankurdotb) [inline x/cheqd/types/keys.go:30]: Is this correct?

**Review by ankurdotb — COMMENTED** [review]
One comment left behind regarding something that seems wrong on namespace. And just to confirm, this is overridden by genesis file namespace if it's present correct?

File: x/cheqd/types/keys.go (hunk 1 above)
Comment (askolesov) [reply x/cheqd/types/keys.go:30]: Yep, it compensates this fix: https://github.com/cheqd/cheqd-node/pull/255/files#diff-abdef39a78c0c284466e9bf9b1c8510681273c75201228b173a6a6a1a466d989L11

**Comment by askolesov** [comment]
This PR seems to be self-contained. Static validation will be implemented in another one.

File: x/cheqd/types/keys.go (hunk 1 above)
Comment (ankurdotb) [reply x/cheqd/types/keys.go:30]: Maybe I'm being dense but explain me one last time tomorrow whether we still need this, given that both testnet and mainnet have this now set via genesis to "testnet" / "mainnet". If I understand correctly, this is internally stored as "testnettestnet" and "mainnetmainnet" but not exposed to the user. Fine. But given that both testnet and mainnet will be running exactly the same version of the software in the future, how will this validation account for that?

**Review by ankurdotb — COMMENTED** [review]
Needs one final discussion tomorrow before closing out

**Comment by askolesov** [comment]
Can we merge this now? I've created a ticket to migrate `did-namespace` key name: https://app.clickup.com/t/6600954/DEV-804.

**Review by ankurdotb — APPROVED** [review]

**Pull request merged**